cmd/babylond/cmd/cmd_test.go

```diff
@@ -23,3 +32,122 @@ func TestInitCmd(t *testing.T) {
 
 	require.NoError(t, svrcmd.Execute(rootCmd, app.BabylonAppEnvPrefix, app.DefaultNodeHome))
 }
+
+func TestStartCmd(t *testing.T) {
+	err := os.RemoveAll(app.DefaultNodeHome)
+	require.NoError(t, err)
+
+	tempApp := app.NewTmpBabylonApp()
+
+	rootCmd := cmd.NewRootCmd()
+	rootCmd.SetArgs([]string{
+		"init",
+		"app-test",
+		fmt.Sprintf("--%s=%s", cli.FlagOverwrite, "true"),
+		fmt.Sprintf("--%s=%s", "bls-password", "testpassword"),
+	})
+
+	require.NoError(t, svrcmd.Execute(rootCmd, app.BabylonAppEnvPrefix, app.DefaultNodeHome))
+
+	blsKeyFile := signer.DefaultBlsKeyFile(app.DefaultNodeHome)
+	blsPasswordFile := signer.DefaultBlsPasswordFile(app.DefaultNodeHome)
+	require.FileExists(t, blsKeyFile, "BLS key file should exist after init")
+	require.FileExists(t, blsPasswordFile, "BLS password file should exist after init")
+
+	keyringDir := filepath.Join(app.DefaultNodeHome, "keyring-test")
+	err = os.MkdirAll(keyringDir, 0o755)
+	require.NoError(t, err)
+
+	kb, err := keyring.New(sdk.KeyringServiceName(), keyring.BackendTest, app.DefaultNodeHome, nil, tempApp.AppCodec())
+	require.NoError(t, err)
+
+	keyInfo, _, err := kb.NewMnemonic(
+		"validator",
+		keyring.English,
+		sdk.FullFundraiserPath,
+		keyring.DefaultBIP39Passphrase,
+		hd.Secp256k1,
+	)
+	require.NoError(t, err)
+
+	addr, err := keyInfo.GetAddress()
+	require.NoError(t, err)
+
+	prepareCmd := cmd.NewRootCmd()
```

Comment:
not sure why we need this and below? the testing scenario is for rpc node runner, right?

Reply:
Yeah but we still need a valid chain genesis and a running chain to test the `start` command otherwise it fails if the initial setup is not correct.